garage/tf/baselines/deterministic_mlp_baseline_with_model.py

```diff
@@ -0,0 +1,71 @@
+"""A value function (baseline) based on a GaussianMLP model."""
+import numpy as np
+
+from garage.misc.overrides import overrides
+from garage.np.baselines import Baseline
+from garage.tf.regressors import DeterministicMLPRegressorWithModel
```

Comment:
do the Deterministic* primitives all work with both continuous and discrete data?

I ask because Gaussian/Categorical/DeterministicThing all make sense as names, unless there needs to be two different deterministic primitives (discrete and continuous). The naming scheme so far is not very consistent. I'd like to pick one going forward and stick with it.

Reply:
I think a `MLP` would always be continuous (since output value is a real value), and if we wrap the `MLP` in a primitive to discretize it, we would call that discrete (e.g. `DiscreteMLPPolicy`/`DiscreteMLPQFunction`). I agree we should not have `Deterministic` and have `Discrete` and `Continuous` instead.

Reply:
alright as long as you're thinking about it. just pick a rule and be consistent going forward. what are you going to name things in this PR?

Reply:
This PR refactor the existing `DeterministicMLPBaseline` which is continuous, I think we should call it `ContinuousMLPBaselineWithModel`?

Reply:
SGTM

Reply:
I agree Continuous is a better name